projects/chaos/src/relay_retry_service.c

```diff
@@ -0,0 +1,59 @@
+#include "relay_retry_service.h"
+
+#include "relay_retry_service.h"
+
+#include "chaos_events.h"
+#include "event_queue.h"
+#include "relay_id.h"
+
+#define RELAY_RETRY_SERVICE_BACKOFF_MS 10
+
+static RelayRetryServiceStorage *s_storage;
+
+StatusCode relay_retry_service_init(RelayRetryServiceStorage *storage) {
+  if (storage == NULL) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+  s_storage = storage;
+  s_storage->max_retries = 0;
+  for (RelayId i = 0; i < NUM_RELAY_IDS; i++) {
+    s_storage->relays_curr_retries[i] = 0;
+  }
+  return STATUS_CODE_OK;
+}
+
+StatusCode relay_retry_service_update(const Event *e) {
+  if (s_storage == NULL || e == NULL) {
+    return status_code(STATUS_CODE_INVALID_ARGS);
+  }
+
+  if (e->id == CHAOS_EVENT_MAYBE_RETRY_RELAY) {
+    if (e->data >= NUM_RELAY_IDS) {
+      return status_code(STATUS_CODE_INVALID_ARGS);
+    }
+    if (s_storage->max_retries != RELAY_RETRY_SERVICE_UNLIMITED_ATTEMPTS &&
```

Comment:
Couldn't you merge these two conditionals?

Reply:
I wanted to not overflow. I refactored it to look nicer though.